examples/settings/settings_main.c

```diff
@@ -0,0 +1,276 @@
+ * apps/examples/settings/settings_main.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/mount.h>
+
+#include "system/settings.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * settings_main
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  int                  ret;
+  int                  fd;
+  enum settings_type_e stype;
+  char                 path[50];
+  char                 *teststr = "";
+  int                  testval = 0x5aa5;
+  char                 readstr[CONFIG_SETTINGS_VALUE_SIZE];
+  enum storage_type_e  storage_type = STORAGE_BINARY;
+  bool                 flag_present = false;
+  struct stat          sbuf;
+
+  ASSERT(0);
+  if ((argc < 1) || *argv[1] == 0 || *(argv[1] + 1) == 0)
+    {
+      goto print_help;
+    }
+
+  if (*argv[1] == '-')
```

Comment:
```suggestion
  if (argc > 1 && *argv[1] == '-')
```
Check first that the argument actually exists?

Reply:
It is used in a variety of apps in the repo and seems to work though?